### PR TITLE
Allow updating search attributes from schedule updates

### LINKED ISSUE #314: [Feature Request] Support schedule search attribute update
### Describe the solution you'd like

See https://github.com/temporalio/features/issues/512 and the Go implementation at https://github.com/temporalio/sdk-go/issues/1561

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Added ability to update the SAs for schedules
Also updated Core, largely to eliminate some compiler warnings from new Rust version

## Why?
Parity

## Checklist

1. Closes https://github.com/temporalio/sdk-dotnet/issues/314
2. How was this tested:
Added tests

3. Any docs updates needed?

### CODE REVIEW

**Comment by Sushisource** [comment]
I looked at the Go impl while doing this and I think the language here needs to be changed: https://github.com/temporalio/sdk-go/pull/1562/files#diff-03e17ff1d7fdefff7ba8a41ea26b6f3a3135e4b084e58cef60574e60bfd9a9f2R493

It is distinctly _not_ an upsert operation. It is straight-up replacement. An upsert would preserve unchanged attributes if they are not included in the new collection, but they are wiped out unless you manually copy the existing attributes.

File: src/Temporalio/Bridge/src/runtime.rs [hunk 1]
```diff
@@ -54,14 +54,6 @@ pub struct ForwardedLog {
     fields_json: Arc<Mutex<Option<String>>>,
 }
 
-#[repr(C)]
-pub struct ForwardedLogDetails {
```
Comment (Sushisource) [inline src/Temporalio/Bridge/src/runtime.rs:58]: This was unused

**Comment by Quinn-With-Two-Ns** [comment]
Yeah would be in favor of fixing the Go SDK docs as well

File: src/Temporalio/Client/Schedules/ScheduleUpdate.cs [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+    /// <li>If present but empty, the search attributes will be cleared.</li>
+    /// <li>If present and non-empty, all search attributes will be updated to the provided
+    ///     collection. This means existing attributes which do not exist in the provided collection
+    ///     will be wiped out. You can copy attributes from the existing <see
+    ///     cref="ScheduleDescription"/> to avoid this.</li>
+    /// </ul>
+    /// </param>
+    public record ScheduleUpdate(
+        Schedule Schedule,
+        SearchAttributeCollection? SearchAttributes = null);
```
Comment (cretz) [inline src/Temporalio/Client/Schedules/ScheduleUpdate.cs]:
```suggestion
        SearchAttributeCollection? TypedSearchAttributes = null);
```

This matches the name on `ScheduleOptions` when creating. I considered accepting `ScheduleOptions` when the inevitable memo update comes about, but it seemed more bad than good.

File: src/Temporalio/Client/TemporalClient.Schedules.cs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Google.Protobuf;
```
Comment (cretz) [inline src/Temporalio/Client/TemporalClient.Schedules.cs]:
```suggestion
```

Comment (Sushisource) [reply src/Temporalio/Client/TemporalClient.Schedules.cs]: Stupid ide

**Review by cretz — APPROVED** [review]

**Pull request merged**